eslint.config.js

```diff
@@ -0,0 +1,810 @@
+import { fixupPluginRules } from '@eslint/compat';
+import typescriptEslint from '@typescript-eslint/eslint-plugin';
+import tsParser from '@typescript-eslint/parser';
+import eslintConfigPrettier from 'eslint-config-prettier';
+import jest from 'eslint-plugin-jest';
+import jestDom from 'eslint-plugin-jest-dom';
+import node from 'eslint-plugin-node';
+import react from 'eslint-plugin-react';
+import reactHooks from 'eslint-plugin-react-hooks';
+import sonarjs from 'eslint-plugin-sonarjs';
+import testingLibrary from 'eslint-plugin-testing-library';
+import globals from 'globals';
+
+export default [
+  {
+    ignores: ['.github', 'eslint.config.js', 'rollup.config.js', 'coverage', 'dist', 'lib']
+  },
+  {
+    name: 'common',
+    files: ['{src,test,website}/**/*.{js,ts,tsx}', 'vite.config.ts'],
+
+    plugins: {
+      node: fixupPluginRules(node),
+      react,
+      'react-hooks': fixupPluginRules(reactHooks),
+      jest,
+      'jest-dom': jestDom,
+      sonarjs,
+      '@typescript-eslint': typescriptEslint,
+      'testing-library': fixupPluginRules(testingLibrary)
+    },
+
+    linterOptions: {
+      reportUnusedDisableDirectives: true
+    },
+
+    languageOptions: {
+      parser: tsParser,
+      ecmaVersion: 'latest',
+      sourceType: 'module',
+
+      parserOptions: {
+        jsxPragma: null,
+        project: './tsconfig.json',
+        lib: ['ESNext'],
```

Comment:
https://typescript-eslint.io/packages/parser/#lib

>If you provide `parserOptions.project`, you do not need to set this, as it will be automatically detected from the compiler.

```suggestion
```